schema.ts

```diff
@@ -72,10 +78,52 @@ type Methods<
   >;
 };
 
+type ContextMethods<
+  S extends ContextService,
+  Def extends Record<string, unknown> = Record<string, never>
+> = {
+  [K in keyof S]?: MethodDetails<
+    Parameters<S[K]>[1],
+    Awaited<ReturnType<S[K]>>,
+    Def
+  >;
+};
+
 interface Logger {
   error: (str: string) => void;
 }
 
+export function contextServiceWithSchema<
+  S extends ContextService,
+  Def extends Record<string, unknown> = Record<string, never>
+>(
```

Comment:
Needs some docs around this

Reply:
Updated the example in the docs to show this instead